sdk/containerregistry/azure-containerregistry/samples/async_samples/sample_create_client_async.py

```diff
@@ -23,31 +23,65 @@
 from dotenv import find_dotenv, load_dotenv
 import os
 
+from azure.identity import AzureAuthorityHosts
+
 
 class CreateClients(object):
     def __init__(self):
         load_dotenv(find_dotenv())
 
+    def get_authority(self, endpoint):
+        if ".azurecr.io" in endpoint:
+            return AzureAuthorityHosts.AZURE_PUBLIC_CLOUD
+        if ".azurecr.cn" in endpoint:
+            return AzureAuthorityHosts.AZURE_CHINA
+        if ".azurecr.us" in endpoint:
+            return AzureAuthorityHosts.AZURE_GOVERNMENT
+        raise ValueError("Endpoint ({}) could not be understood".format(endpoint))
+
+    def get_credential_scopes(self, authority):
```

Comment:
We should not publish this as a sample. We do not want customers to have this code in their applications.

Reply:
Is there a best practices for demonstrating how to authenticate when in a sovereign cloud? or another sample that could demonstrate this

Reply:
The intent is this: https://github.com/Azure/azure-sdk/blob/ba2da3c8d6273c77d24b947ea6352970c9bee47f/docs/python/clouddiscover.md

Reply:
We've also decided not to use this approach because if customers have custom endpoints (vanity URLs) in national clouds, this would not give the desired result.  That's why we're letting them pass the scope directly (in this version) -- it's because we need this to know which cloud we're in if they have a custom endpoint.

Reply:
@annelo-msft is the "this approach" you're referring to the approach demonstrated in the samples or in the clouddiscover.md document?

Reply:
In the sample ... I must not have refreshed because I didn't see @johanste's second comment.  Apologies!

Reply:
Removed all of this, this will need to be addressed in the future